### DISCUSSION TITLE
Welcome to game_cheetah Discussions!

### DISCUSSION THREAD

**Discussion by mkrueger** [mkrueger/game_cheetah#1] [Announcements]
I'll need to add a freeze function and then I would call it ready.

The tool should be simple to use but possible extensions would be to set the data type to search ((u)int8/16/32/64, f32/64) - however I don't know if it's really worth it.

Esp. float values are often displayed as rounded numbers.

**Comment by mkrueger** [comment]
freeze function and some ui improvements are done, multiple searches work. Time for release :).

**Comment by shoutmarble** [comment]
I like Game Cheetah because it does the "1" thing that I want a
Game Cheetah to do!  And that is to make game cheating easy!

```
0.3.2 Release
Latest
Aug 3, 2023
```

One note, is I would switch from GTK4 to [iced](https://github.com/iced-rs/iced).

I had problem compiling git main with Rust GTK.  I switched to nightly and compiled GTK4
with gvsbuild successfully.

**Reply by mkrueger** [reply]
Never had GTK4 in game cheetah - was a dep of egui. I used egui. I started with iced but switched to egui at some point. However I lately switched back to iced. 

Rust makes it very easy to switch UI libs IMO.

**Reply by shoutmarble** [reply]
I'm very excited about Rust also.  I've dabbled into embedding a WASM binary into my Rust program and invoking it via WASI and a WASM runtime into my Rust executable which is brain melting aweome.

Sort of like embedding a docker "image" into a Docker Runtime... or a JVM running in your program...or a Node running in your program...Javascript running in the browser... etc.

The thing in rust is that the data structures to handle mutable and atomic and the shuttling data mechanisms for them is an amazing deeply computer "sciencey" area that is really really hard to pull together.

That and AI is making productivity even in Rust absolutely gonzo.

I have a few Rust programs I have in the pipeline...I kind of want to make own "digital paper" like the type used in digital pens (livescribe) --there are just too many missing pieces of making a good open source digital pen yet.

I've tackled a modest ESP32 with a 1.5" TFT with Rust was absolutely blown away with the possibilities of a $20 micro-controller with Wifi and a digital screen.  I have no talent for that area of programming...or the other with making digital paper.

[Olorin](https://github.com/Squalr/Olorin) We may be ending the world of game cheats with AI created "game" worlds in first-person.

Thank you for the great software.

**Reply by mkrueger** [reply]
I made some rust tools - I love it too. Programming languages/compilers etc. is basically the stuff I've been working on for ages and from that perspective rust is even better. It's nearly perfect. Would've chosen another syntax here and there but overall it's beyond awesome :).

This is the language that was badly needed for computer science IMO - finally stable & fast software is possible and a realistic perspective. I never tried rust in an embedded project - have some little arduino/esp solutions in my smart home - in the next one I'll try rust too.
